importers/ableroad.com/easy_importer.py

```diff
@@ -0,0 +1,226 @@
+#!/home/liu1ee/apps/virtualenv/bin/python
+from lxml import html
+import requests
+import os.path
+import re
+import csv
+import urllib2
+import logging
+import time
+
+def getlatlng(js):
+   """
+   a sample js string lookes like this
+   #//debug stuff maphidden = 0;
+       if (!maphidden) {
+                createSearchMarker('20. Craft Heads Brewing Company','<label style="display:none">phone and address</label>89 University Avenue W<br/>Windsor, ON N9A 5N8<br/>Canada<br/>+1-226-246-3925','','ableroad.com/edit.php?index=20&amp;newID=craft-heads-brewing-company-windsor&amp;s=&amp;s1=windsor,ontario&amp;action=write',42.317252218728,-83.039797096553,20 - 1,1);
+                     }
+   """
+   js = js.encode('ascii','ignore') # unicode to ascii
+   reg = r'\([^)]+\)'
+   m = re.findall(reg, js)
+   return re.split(',\s*',m[1])[-4:-2]
+def getname(s):
+    if not len(s):
+        return ''
+    reg = r'\d+\.\s*(?P<name>(\w|\s)+)'
+    m = re.match(reg,s[0])
+    if not m:
+        return ''
+    return m.group('name')
+
+def getcategory(s):
+    if not len(s):
+        return ''
+    return re.split(r':\s*',s[0])[1]
+
+def getdistance(s):
+    if not len(s):
+        return ''
+    reg = r'(?P<dis>\d+\.?\d*)'
+    m = re.search(reg,s[0])
+    if not m:
+        return ''
+    #conver mile to km
+    dis = m.group('dis')
+    dis = float(dis) * 1.60934
+    dis = str(round(dis, 2))
+    return dis
+
+def getneighborhood(s):
+    if not len(s):
+        return ''
+    return re.split(r':\s*',s[0])[1]
+
+def getableroad(s):
+    if not len(s):
+        return ''
+    reg = r'(?P<ablerating>\d+\.?\d*)'
+    m = re.search(reg,s[0])
+    if not m:
+        return ''
+    return m.group('ablerating')
+
+def getyelp(s):
+    if not len(s):
+        return ''
+    reg = r'(?P<yelp>\d+\.?\d*)'
+    m = re.search(reg,s[0])
+    if not m:
+        return ''
+    return m.group('yelp')
+
+def getstreet(s):
+    if not len(s):
+        return ''
+    return s[0]
+
+def getcity(s):
+    if len(s) < 1:
+        return ''
+    s = ' '.join(s)
+    m = re.search(r'(?P<city>[a-zA-Z]+),\s*', s)
+    if not m:
+        return ''
+    return m.group('city')
+
+def getstate(s):
+    if not len(s):
+        return ''
+    s = ' '.join(s)
+    m = re.search(r'(?P<city>[a-zA-Z]+),\s*(?P<state>[a-zA-Z]+)', s)
+    if not m:
+        return ''
+    return m.group('state')
+
+def getpostcode(s):
+    if not len(s):
+        return ''
+    s = ' '.join(s)
+    m = re.search(r'(?P<city>[a-zA-Z]+),\s*(?P<state>[a-zA-Z]+)\s+(?P<post>(\d{5,5})|(\w{3}\s\w{3}))', s)
+    if not m:
+        return ''
+    return m.group('post')
+
+def getphone(s):
+    if len(s) < 1:
+        return ''
+    if not re.search(r'\d{2,}-\d+',s[-1]):
+        return ''
+    return s[-1]
+
+def extract_info(dombus):
+    row = []
+    name = dombus.xpath('.//a[@class="titlelink"]/text()')
```

Comment:
I could not find a cssselector that could select node form current node, rather than root node.
name = locationdiv.xpath('.//a[@class="titlelink"]/text()') //xpath it will select the content in the location div node, instead of html root document.
return ['1. adb']
name = locationdiv.cssselect('a[@class="titlelink"]') will select all title link in the document rather than locationdiv title link.
return ['1. adb','2. adfa','3. adfa','4. adfa']
any idea for this issue.